### ISSUE TITLE
Allow to use other drives for distro installation

### ISSUE THREAD

**Issue by sr229** [microsoft/WSL#5739]
**Describe the solution you'd like**
Currently, when we install from Microsoft store, we use C:\ by default, however, we have the ability to select whatever drive we can use if we install manually (sacrificing having to do first time user setup however). I think we should have the distro launchers support selecting a host drive instead of relying on just C:\,  this is useful for multi-drive laptops where the SSD is only used as a system drive, and are expected to use the hard drive.

**Describe alternatives you've considered**
`wsl --import` or `wsl --install` works fine but it lacks the first-time user install experience.

**Comment by Biswa96** [comment]
Packages/Programs from Windows Store runs in a jail/container/sandbox like environment. If Windows Store allows to install in another drive it would be a sweet spot for you-know-who. You can grab the .tar.gz file from Windows Store installed folder and use it to install in another drive.

**Comment by sr229** [comment]
> Packages/Programs from Windows Store runs in a jail/container/sandbox like environment. If Windows Store allows to install in another drive it would be a sweet spot for you-know-who. You can grab the .tar.gz file from Windows Store installed folder and use it to install in another drive.

We already can migrate apps over other drives. Plus DistroLauncher is just an application to unpack or launch the distro installed. I don't see in a security standpoint it would be insecure since we already have the feature set for it.

**Comment by Okesik** [comment]
You can change MS Store install drive in Win10 Settings > System > Storage for Apps or migrate already installed apps to another drive.
If you feel that MS Store is rightfully the worst one there is, there's a feedback button for it.

I don't think this belongs here at all however. You can install your own distro into WSL without relying on MS Store so WSL itself is working as intended.

**Comment by sr229** [comment]
> You can change MS Store install drive in Win10 Settings > System > Storage for Apps or migrate already installed apps to another drive.
> If you feel that MS Store is rightfully the worst one there is, there's a feedback button for it.
> 
> I don't think this belongs here at all however. You can install your own distro into WSL without relying on MS Store so WSL itself is working as intended.

I don't think this is the desirable response here, while I agree to allowing feedback, I think the WSL experience could be far more improved if the installation path of WSL followed on where the DistroLauncher apps for the respective distros are. So far, DistroLauncher distros are stuck only in the C:\ Drive - that's something we can work on in the long run since we already have the feature for it - we're just not readily using it. This concerns WSL as a whole as well since DistroLauncher is actually a part of WSL - it's primary purpose is to launch the supplied distro. As a user who might be just getting into the Linux ecosystem via WSL, you might want to move your stuff as easy as clicking it over at the settings, we want to have that as well for DistroLauncher distros.

**Comment by rubyFeedback** [comment]
I think this would be a good idea - it allows some flexibility. I do not necessarily disagree with Biswa96, but to me I think this is more a usability question than a security question. You can always provide the information given here, use with caution etc... while allowing it still. (A few years ago, or perhaps ... 2 years ago I think, with WSL, I used to reason that the whole customization/installation should be very flexible, and also inform the user what has happened, in regards to distribution-installations. I am hugely forgetful and literally with all the multitasking I do in general, being flexible is great, while also keeping track of what has happened in files and GUIs etc... - so I think this would be a usability suggestion more than a security-related problem set really. That being said, I am mostly really just neutral on this, I am fine either way, just giving my opinion. :-) )
